include/seqan3/core/simd/view_iota_simd.hpp

```diff
@@ -0,0 +1,248 @@
+    counted_simd_iterator operator++(int)
+    {
+        counted_simd_iterator tmp{*this};
+        ++(*this);
+        return tmp;
+    }
+
+    //!\brief Returns the distance between two iterators.
+    difference_type operator-(counted_simd_iterator const & rhs) const
+    {
+        return count_scalar - rhs.count_scalar;
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+
+    //!\brief Tests whether `lhs == rhs`.
+    friend bool operator==(counted_simd_iterator const & lhs, counted_simd_iterator const & rhs) noexcept
+    {
+        return lhs.count_scalar == rhs.count_scalar;
+    }
+
+    //!\brief Tests whether `lhs != rhs`.
+    friend bool operator!=(counted_simd_iterator const & lhs, counted_simd_iterator const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\brief A closed iota view over a simd vector.
+ * \ingroup simd
+ * \implements std::ranges::forward_range
+ * \implements std::ranges::sized_range
+ * \implements std::ranges::common_range
+ * \implements std::ranges::borrowed_range
+ *
+ * \tparam index_t The represented index type; must model seqan3::simd::simd_concept.
+ *
+ * \details
+ *
+ * This view is a lightweight wrapper around a seqan3::detail::counted_simd_iterator pair.
+ * The regular std::views::iota view cannot be used in combination with a simd vector type because the comparison of
+ * two simd vectors does not return a bool but another simd vector type.
+ *
+ * ### View properties
+ *
+ * | Concepts and traits              | `rrng_t` (returned range type) |
+ * |----------------------------------|:------------------------------:|
+ * | std::ranges::input_range         | *guaranteed*                   |
+ * | std::ranges::forward_range       | *guaranteed*                   |
+ * | std::ranges::bidirectional_range | *lost*                         |
+ * | std::ranges::random_access_range | *lost*                         |
+ * | std::ranges::contiguous_range    | *lost*                         |
+ * |                                  |                                |
+ * | std::ranges::viewable_range      | *guaranteed*                   |
+ * | std::ranges::view                | *guaranteed*                   |
+ * | std::ranges::sized_range         | *guaranteed*                   |
+ * | std::ranges::common_range        | *guaranteed*                   |
+ * | std::ranges::output_range        | *lost*                         |
+ * | std::ranges::borrowed_range      | *guaranteed*                   |
+ * | seqan3::const_iterable_range     | *guaranteed*                   |
+ * |                                  |                                |
+ * | std::ranges::range_reference_t   | simd_t                         |
+ *
+ * For more details, see \ref views.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/simd/view_iota_simd.cpp
+ */
+template <simd_concept index_t>
```

Comment:
```suggestion
template <simd_concept simd_t>
```
?

Reply:
but it is the index type and which is constrained to be a simd type. I also don't call it int_t if it is an integer index?

Reply:
true, but I think that `simd` is like `ptr` or `it` something special that we should encode in the type.

How would be `index_simd_t` ?

Reply:
```suggestion
template <simd_concept index_simd_t>
```